### PR TITLE
[DemStructures] Added a tool to write SP data in HDF5 format

### PR DESCRIPTION
**Pull request by ipouplana**
Summary of contents in this PR:

- Python tool to write the data in HDF5
- C++ tool to get lists of DEM member variables
- Two python scripts to post-process the stored data (a bit ad-hoc, but they could be useful)

This can be of course enhanced with other features, but I thought it was important to have it already in the master branch.

### CODE REVIEW

File: applications/DemStructuresCouplingApplication/python_scripts/sp_statistics.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import math
+import matplotlib.pyplot as plt
+import numpy as np
+import h5py
```
Comment (philbucher) [inline applications/DemStructuresCouplingApplication/python_scripts/sp_statistics.py:3]: not sure if applicable; have you considered using the HDF5-Application?

Comment (ipouplana) [reply applications/DemStructuresCouplingApplication/python_scripts/sp_statistics.py:3]: Thanks, I know that we had the possibility of using it but, at least for what we needed, I think this implementation is simple and it's enough.

**Comment by ipouplana** [comment]
@GuillermoCasas could yo take a look at this?

**Review by GuillermoCasas — CHANGES_REQUESTED** [review]
Please just address those easy codacy warnings and I will approve

**Review by GuillermoCasas — APPROVED** [review]

**Pull request merged**